platform/darwin/src/MGLShapeSource.h

```diff
@@ -116,11 +116,14 @@ MGL_EXPORT
 
 /**
  The contents of the source. A shape can represent a GeoJSON geometry, a
- feature, or a collection of features.
+ feature, or a collection of features. 
 
  If the receiver was initialized using `-initWithIdentifier:URL:options:`, this
  property is set to `nil`. This property is unavailable until the receiver is
  passed into `-[MGLStyle addSource:]`.
+ 
+ You can get/set the shapes within a collection via this property. Setting
+ this property must be performed on the application's main thread.
```

Comment:
Given that the setter isn’t background-thread-safe, I’d be surprised if the getter were completely thread-safe. Best to say that this property must be accessed on the main thread and leave it at that.